src/sentry/api/endpoints/project_rules.py

```diff
@@ -49,7 +53,28 @@ def post(self, request, project):
         serializer = RuleSerializer(context={"project": project}, data=request.data)
 
         if serializer.is_valid():
-            rule = serializer.save(rule=Rule())
+            data = serializer.validated_data
+            kwargs = {
+                "name": data["name"],
+                "environment": data.get("environment"),
+                "project": project,
+                "action_match": data["actionMatch"],
+                "conditions": data["conditions"],
+                "actions": data["actions"],
+                "frequency": data.get("frequency"),
```

Comment:
I'm still not sure passing these nodes to the task is a great idea. This sort of thing can break when we upgrade DRF or Django. One option might be to just pass the original json blob from the request through to the task, and just pass it all to the serializer again in the task?

Reply:
@wedamija I am confused what the issue is here. Isn't `kwargs` just a json dictionary?

Reply:
The `validated_data` is an OrderDict so I think we should be okay passing through the values we grab out of it.

Reply:
Yeah, this is fine. I misunderstood something that we discussed earlier.